### PR TITLE
Update Windows Server 2019 to Visual Studio 2019

### PR DESCRIPTION
**Pull request by CyanDevs**
This PR updates Windows Server 2019 to use Visual Studio 2019, which includes an update to cmake 3.20.21032501 and ninja 1.10.2. 

Other fixes:
* Windows Docker images "oetools" are also updated to VS2019.
** Note that this has greatly increased the final uncompressed image size from 16GB to 22GB.
** Reduced image build context size, and to skip installation of Intel drivers as originally intended (To investigate: Intel devices support in Windows containers)
* Ubuntu image users are not deprovisioned when generating the Azure managed image.
* Allow Windows and libcxx tests to run on custom Jenkins agents

### CODE REVIEW

**Comment by CyanDevs** [comment]
bors try

**Review by mingweishih — APPROVED** [review]

**Review by radhikaj — APPROVED** [review]

File: .jenkins/infrastructure/docker/dockerfiles/windows/Dockerfile [hunk 1]
```diff
@@ -9,5 +9,5 @@
 FROM mcr.microsoft.com/windows/servercore:ltsc2019
 
 COPY . oe
-RUN powershell.exe -ExecutionPolicy Unrestricted -NoLogo -NonInteractive -NoProfile -WindowStyle Hidden -File oe\scripts\install-windows-prereqs.ps1 -InstallPath C:\oe_prereqs -LaunchConfiguration SGX1FLC-NoIntelDrivers -DCAPClientType Azure
+RUN powershell.exe -ExecutionPolicy Unrestricted -NoLogo -NonInteractive -NoProfile -WindowStyle Hidden -File oe\scripts\install-windows-prereqs.ps1 -InstallPath C:\oe_prereqs -LaunchConfiguration SGX1FLC-NoIntelDrivers -DCAPClientType None
```
Comment (radhikaj) [inline .jenkins/infrastructure/docker/dockerfiles/windows/Dockerfile:12]: Why did you change this to DCAPClientType None?

Comment (CyanDevs) [reply .jenkins/infrastructure/docker/dockerfiles/windows/Dockerfile:12]: Because this is a container environment, we don't need to install the drivers nor run on an Azure VM. Otherwise, it would try to install both and fail.

File: .jenkins/pipelines/Azure/Windows/Jenkinsfile [hunk 2]
```diff
@@ -10,28 +10,28 @@ try{
...
             testing_stages += [
-                "Win2019 Ubuntu1804 clang-10 RelWithDebInfo Linux-Elf-build LVI":    { tests.windowsLinuxElfBuild('acc-win2019-dcap', '18.04', 'clang-10', 'RelWithDebInfo', 'ControlFlow', 'ON') },
-                "Win2019 Sim RelWithDebInfo Cross Compile LVI ":                     { tests.windowsCrossCompile('acc-win2019', 'RelWithDebInfo', 'ControlFlow', '1', 'ON') },
-                "Win2019 Debug Cross Compile DCAP LVI":                              { tests.windowsCrossCompile('acc-win2019', 'Debug',          'ControlFlow', '0', 'ON') },
-                "Win2019 RelWithDebInfo Cross Compile DCAP LVI FULL Tests":          { tests.windowsCrossCompile('acc-win2019', 'RelWithDebInfo', 'ControlFlow') },
-                "Win2019 RelWithDebInfo Cross Compile DCAP LVI FULL Tests snmalloc": { tests.windowsCrossCompile('acc-win2019', 'RelWithDebInfo', 'ControlFlow', '0', 'OFF', ['-DUSE_SNMALLOC=ON']) }
+                "Win2019 Ubuntu1804 clang-10 RelWithDebInfo Linux-Elf-build LVI":    { tests.windowsLinuxElfBuild(params.WS2019_DCAP_CFL_LABEL, params.UBUNTU_1804_NONSGX_LABEL, 'clang-10', 'RelWithDebInfo', 'ControlFlow', 'ON') },
```
Comment (radhikaj) [inline .jenkins/pipelines/Azure/Windows/Jenkinsfile:30]: why do you have a CFL label?

**Review by radhikaj — APPROVED** [review]

File: .jenkins/infrastructure/docker/dockerfiles/windows/Dockerfile (hunk 1 above)
Comment (radhikaj) [reply .jenkins/infrastructure/docker/dockerfiles/windows/Dockerfile:12]: Do we not use the same containers for testing?

File: .jenkins/pipelines/Azure/Windows/Jenkinsfile (hunk 2 above)
Comment (CyanDevs) [reply .jenkins/pipelines/Azure/Windows/Jenkinsfile:30]: This is so we can distinguish between ICX and CFL agents while also parameterizing the agent labels so we can run these on E2E agents as well (since for E2E agents we use the same labels as their normal counterpart, but with an "-e2e" suffix).

File: .jenkins/infrastructure/docker/dockerfiles/windows/Dockerfile (hunk 1 above)
Comment (CyanDevs) [reply .jenkins/infrastructure/docker/dockerfiles/windows/Dockerfile:12]: We don't use Windows containers for tests yet, so this is not an issue
